### DISCUSSION TITLE
nextjs application unable to read .env file on production

### DISCUSSION THREAD

**Discussion by harshitaGoswamii** [vercel/next.js#52017] [Help]
### Summary

**Issue:** 
During testing , my application is not taking any images or css and displaying only the HTML structure , on clicking any URL it is redirecting to the Local URL . Below is the image for the same.
![image](https://github.com/vercel/next.js/assets/135982292/190fd2fa-c90f-440e-84ab-99e799503000)

**My solution was**
I made an env file with my testing url and wrote a functionality in to replace the local host URLs with your testing URL

` import env from "../../env"
//appending base url with path url

const baseURL = env.baseUrl;
const footerWithBaseUrl = footerData.sections.map((item) => ({
  ...item,  url: `${baseURL}${item.url}`,`
  
  How can I fix this issue?

**Comment by icyJoseph** [comment]
Are you deploying on vercel? is the .env file included in the git history? if you are deploying to Vercel, you are better off including the env vars in the Vercel Dashboard for your project

**Reply by harshitaGoswamii** [reply]
No, Ubuntu server

**Reply by icyJoseph** [reply]
I have read your post again. Could you describe the problem with code?

Like, why is the css not loading, related to the env vars? Are the env vars available during build time? Public env vars are inlined at build time. So if they weren't present then, they won't work when you run the app.

**Comment by BrunoFerreira95** [comment]
Are you using the "NEXT_PUBLIC_" pre-fix in the key to share with the Browser or
Are you made an env private?

**Reply by harshitaGoswamii** [reply]
Yes, I used this prefix 
`NEXT_PUBLIC_API_URL=http://testing-url`

**Comment by vignesh-gupta** [comment]
I think instead of importing. Can you use `process.env.API_URL`. And While running just the logs should have loaded env.
